gax-java/gax/src/test/java/com/google/api/gax/rpc/WatchdogTest.java

```diff
@@ -154,6 +154,31 @@ public void testTimedOutBeforeStart() throws InterruptedException {
     assertThat(error).isInstanceOf(WatchdogTimeoutException.class);
   }
 
+  @Test
+  public void testTimedOutBeforeResponse() throws InterruptedException {
+    MockServerStreamingCallable<String, String> autoFlowControlCallable =
+        new MockServerStreamingCallable<>();
+    AutoFlowControlObserver<String> downstreamObserver = new AutoFlowControlObserver<>();
+
+    autoFlowControlCallable.call("request", watchdog.watch(downstreamObserver, waitTime, idleTime));
+    MockServerStreamingCall<String, String> call1 = autoFlowControlCallable.popLastCall();
+
+    clock.incrementNanoTime(idleTime.toNanos() + 1);
+    watchdog.run();
+    assertThat(downstreamObserver.done.isDone()).isFalse();
+    assertThat(call1.getController().isCancelled()).isTrue();
+    call1.getController().getObserver().onError(new CancellationException("cancelled"));
+
+    Throwable actualError = null;
+    try {
+      downstreamObserver.done.get();
+    } catch (ExecutionException e) {
+      actualError = e.getCause();
+    }
+    assertThat(actualError).isInstanceOf(WatchdogTimeoutException.class);
+    assertThat(actualError.getMessage()).contains("waiting for next response");
```

Comment:
qq, why is this a WatchdogTimeoutException and not the CancellationException above?

Reply:
Calling onError on the observer will call onErrorImpl in the WatchdogStream (line 299). And if the stream is already cancelled by watchdog, the error on this watchdog stream will be the WatchdogException (line 348).

Reply:
> the error on this watchdog stream will be the WatchdogException

Gothca. I took a look at the lines again and just have a clarifying quesiton. The WatchDogException you linked above is from the `cancelIfStale()` call, right? If the stream was already cancelled from watchdog, why is `cancelIfStale()` still run? I think it should have been removed from the openstreams from the `onErrorImpl` call right?

Reply:
watchdog is a runnable that calls cancelIfStale (line 131). Which in this test is called on line 167 `watchdog.run()`

Reply:
This was resolved in an offline discussion with Mattie. 

The follow question was how the CancellationException became a WatchdogTimeoutException. It's from this line: https://github.com/googleapis/sdk-platform-java/blob/8dc7c52e63f23e9be6dde82aa54f9f077a80b2f8/gax-java/gax/src/main/java/com/google/api/gax/rpc/Watchdog.java#L300-L303